engine/immediate_task_conrrent_processor.go

```diff
@@ -426,3 +425,25 @@ func (w *immediateTaskConcurrentProcessor) composeHttpError(
 
 	return statusCode, details, fmt.Errorf("statusCode: %v, errMsg: %w, responseBody: %v", statusCode, err, responseBody)
 }
+
+func (w *immediateTaskConcurrentProcessor) processLocalQueueMessagesTask(
+	ctx context.Context, task persistence.ImmediateTask,
+) error {
+	resp, err := w.store.ProcessLocalQueueMessages(ctx, persistence.ProcessLocalQueueMessagesRequest{
+		TaskShardId:  task.ShardId,
+		TaskSequence: task.GetTaskSequence(),
+
+		ProcessExecutionId: task.ProcessExecutionId,
+
+		Messages: task.ImmediateTaskInfo.LocalQueueMessageInfo,
+	})
+	if err != nil {
+		return err
+	}
+
+	// TODO: need a better way to define the NotifyNewImmediateTasks method
+	if resp.HasNewImmediateTask {
+		w.taskNotifier.NotifyNewImmediateTasksNoRequest(task.ShardId)
```

Comment:
`NotifyNewImmediateTasksNoRequest` is probably not needed. Essentially, the `NotifyImmediateTasksRequest` only requires `shardId`. All other fields are optional. I think we can get processExecutionId here. You can ignore other fields if they are not available. (The namespace/processId are  mainly  for future support NoSQL because the partitionKey will be namespace+processId, which is not needed for SQL)